core/tls-websocket-server/src/connection_id_generator.rs

```diff
@@ -0,0 +1,74 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+#[cfg(feature = "sgx")]
+use std::sync::SgxRwLock as RwLock;
+
+#[cfg(feature = "std")]
+use std::sync::RwLock;
+
+use crate::{error::WebSocketError, WebSocketResult};
+
+pub type ConnectionId = usize;
+
+/// Trait to generate IDs (nonce) for websocket connections.
+pub trait GenerateConnectionId {
+	fn next_id(&self) -> WebSocketResult<ConnectionId>;
+}
+
+pub struct ConnectionIdGenerator {
+	current_id: RwLock<ConnectionId>,
+}
+
+impl Default for ConnectionIdGenerator {
+	fn default() -> Self {
+		Self { current_id: RwLock::new(10) }
+	}
+}
+
+impl GenerateConnectionId for ConnectionIdGenerator {
+	fn next_id(&self) -> WebSocketResult<ConnectionId> {
```

Comment:
There is not - in order to make this safe, quite a bit of logic would be needed. In case we reach max, we would need to find an ID that is within the range and not in use anymore.

`usize` on a 64bit system is 2^64-1. To put that in perspective: If we created a new web-socket connection every millisecond, we could run for 584'542'046 years until we run out of IDs. So I think we should be safe for a first approach 😄

Reply:
I see, that makes sense. Still - how about using [saturating_add](https://docs.rs/num-traits/0.2.14/num_traits/ops/saturating/trait.SaturatingAdd.html) instead?

Reply:
Maybe CLion does this wrong, but as soon as you import something from `core`, it will add the `extern crate` statement as well.

Reply:
I can change it to a `saturating_add`, although I don't see what good it does. In case we really have an overflow, `saturating_add` will break the code regardless. If we really want to catch an overflow, we need to to a `checked_add`, if that fails, re-initialize with the minimum ID (which is 10, so `wrapping_add` would not work either).

Reply:
(changed it to `checked_add` with a reset to the minimum ID)